Dockerfile.okd

```diff
@@ -1,9 +1,9 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.18 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.23-openshift-4.19 AS builder
```

Comment:
@dtfranz 

The project is using go 1.22 
So, that will not work.
We would need to upgrade the go version OR use 1.22-openshift-4.19


c/c @grokspawn @perdasilva

Reply:
I've bumped to 1.23 (cc @dtfranz)

Reply:
Bumping go.mod to 1.23 is fine, but it is also fine (and often the correct thing to do) to use a go 1.23 toolchain to build a project with `go 1.22` because it makes the project compatible for others that import your project who have reasons to stick to a lower go version in their project.